packages/api/src/index.js

```diff
@@ -0,0 +1,41 @@
+import { Router } from 'itty-router'
+import { carHead, carGet, carPut, carPost } from './car.js'
+
+const router = Router()
+
+router.get('/car/:cid', carGet)
+router.head('/car/:cid', carHead)
+router.put('/car/:cid', carPut)
+router.post('/car', carPost)
+
+router.get('/', () => {
+  return new Response(
+    `
+<body style="font-family: -apple-system, system-ui">
+  <h1>📦</h1>
+  <p>try
+    <a href='/car/bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy'>
+      /car/bafkreigh2akiscaildcqabsyg3dfr6chu3fgpregiymsck7e7aqa4s52zy
+    </a>
+  </p>
+</body>`,
+    {
+      status: 200,
+      headers: {
+        'content-type': 'text/html; charset=UTF-8',
+      },
+    }
+  )
+})
+
+router.all('*', () => new Response('Not Found.', { status: 404 }))
+
+async function serverError(error) {
+  return new Response(error.message || 'Server Error', {
+    status: error.status || 500,
+  })
+}
+
+addEventListener('fetch', (event) =>
+  event.respondWith(router.handle(event.request, {}, event).catch(serverError))
```

Comment:
This signature passes things through to the handlers in a way that mimics the pattern that will be in place when cloudflare fully supports sending them am ESM module that exports a handler as per https://developers.cloudflare.com/workers/cli-wrangler/configuration#modules 
```js
import html from './index.html'
export default {  
// * request is the same as `event.request` from the service worker format  
// * waitUntil() and passThroughOnException() are accessible from `ctx` instead of `event` from the service worker format  
// * env is where bindings like KV namespaces, Durable Object namespaces, Config variables, and Secrets  
// are exposed, instead of them being placed in global scope.  
  async fetch(request, env, ctx) {
    const headers = { 'Content-Type': 'text/html;charset=UTF-8' }
    return new Response(html, { headers })
  }}
```

but i think it'll be annoying that we can't just copy paste handlers from nft.storage, which just pass in the `event`, so am going to fix thist PR to pass the request around in the same way.

cc @alanshaw

Reply:
oh, the router `itty-router` expects the first argument to be the `event.request` so, even if i switch it i'm gonna need to update the method signatures on handlers from nft.storage from `handle(event)` to `handle(request, event)`.

Reply:
> The router handle expects only the request itself, but passes along any additional params to the handlers/middleware. For example, to access the event itself within a handler (e.g. for event.waitUntil()), you could simply do this:
```
addEventListener('fetch', event =>
  event.respondWith(router.handle(event.request, event))
)
```
– https://github.com/kwhitley/itty-router#a-few-notes-about-this

Reply:
From conversation with @alanshaw we have a mild preference in favour of switching to bundling the worker as esm (which the wrangler docs suggest will be the recommended format in the future). It's assumed it will be easier to test than a _sort of service-worker_, with magic globals for secrets and other cloudflare features. In esm mode, the env is passed to the fetch function explicitly with all the magic properties on.

https://developers.cloudflare.com/workers/cli-wrangler/configuration#modules